src/Forte.Optimizely.ContentUsage/Api/Features/ContentType/ContentTypeUsagesRepository.cs

```diff
@@ -86,17 +91,79 @@ INNER JOIN
-		            Count = (int)dbDataReader["Count"]
-	            };
-
-	            contentTypeUsageCounters.Add(contentTypeUsageCounter);
+                contentTypeUsageCounters.Add(contentTypeUsageCounter);
             }
 
             return contentTypeUsageCounters;
         }));
     }
+
+    public async Task<IEnumerable<UsagePage>> GetUsagePages(BlockType contentType, CancellationToken cancellationToken)
+    {
+        var executor = _dataExecutorAccessor();
+        return await executor.ExecuteAsync(async () =>
+        {
+            var command = executor.CreateCommand();
+            command.CommandText = @"
+DECLARE @Usages TABLE (ContentID int, WorkID int, [Name] nvarchar(255), LanguageBranch nchar(17))
+INSERT @Usages EXEC netPageTypeGetUsage @PageTypeID,@OnlyPublished=1
+
+;WITH OwnerContent_CTE AS (
+	SELECT U.ContentID, tblContentSoftlink.fkOwnerContentID AS OwnerContentId
+	FROM @Usages U
+	INNER JOIN tblContent ON tblContent.pkID = U.ContentId
+	INNER JOIN tblContentSoftlink ON tblContentSoftlink.fkReferencedContentGUID = tblContent.ContentGUID
+	WHERE tblContentSoftlink.LinkType=1
+
+	UNION ALL
+
+	SELECT OwnerContent_CTE.ContentID, tblContentSoftlink.fkOwnerContentID AS OwnerContentId
+	FROM OwnerContent_CTE
+	INNER JOIN tblContent ON tblContent.pkID = OwnerContent_CTE.OwnerContentID
+	INNER JOIN tblContentSoftlink ON tblContentSoftlink.fkReferencedContentGUID = tblContent.ContentGUID
+	WHERE tblContentSoftlink.LinkType=1
+		AND tblContent.ContentType=1
+)
+SELECT OwnerContent_CTE.ContentID, OwnerContent_CTE.OwnerContentID, tblContentType.Name AS OwnerContentName
+FROM OwnerContent_CTE
+INNER JOIN tblContent ON tblContent.pkID = OwnerContent_CTE.OwnerContentId
+INNER JOIN tblContentType ON tblContentType.pkID = tblContent.fkContentTypeID
+INNER JOIN tblWorkContent ON tblWorkContent.fkContentID = OwnerContent_CTE.OwnerContentId
+WHERE tblContent.ContentType = 0
+AND tblWorkContent.Status=4
+AND (tblWorkContent.StopPublish IS NULL OR tblWorkContent.StopPublish > @NowTime)
+";
+
+            command.Parameters.Add(executor.CreateParameter("PageTypeID", contentType.ID));
+            command.Parameters.Add(executor.CreateParameter("NowTime",
+                _databaseDateTimeHandler.ConvertToDatabase(DateTime.Now)));
+
+            var usagePages = new List<UsagePage>();
+            await using var dbDataReader = await command.ExecuteReaderAsync(cancellationToken);
+            while (await dbDataReader.ReadAsync(cancellationToken))
+            {
+                var contentTypeUsageCounter = new UsagePage
+                {
+                    ContentLink = new ContentReference((int)dbDataReader["OwnerContentID"]),
+                    PageType = (string)dbDataReader["OwnerContentName"]
+                };
+
+                usagePages.Add(contentTypeUsageCounter);
+            }
+
+            return usagePages;
+        });
+    }
+}
+
+public class UsagePage
```

Comment:
I see another class with the same name. So maybe should you change this class to UsagePageDb?